DistroLauncher-Tests/DistroLauncher-Tests/IniFindValueTests.cpp

```diff
@@ -0,0 +1,116 @@
+default = root
+[boot]
+command = /usr/libexec/wsl-systemd
+)"};
+            std::wstringstream fakeFile{buffer};
+            ASSERT_FALSE(ini_find_value(fakeFile, L"boot", L"command", L"/usr/libexec/wsl-systemd"));
+            ASSERT_FALSE(ini_find_value(fakeFile, L"user", L"default", L"root"));
+        }
+
+        TEST(IniFindValueTests, CommentedLineCommandMustResultFalse)
+        {
+            std::wstring buffer{LR"(# This mimics the syntax of /etc/wsl.conf file with comments.
+[user]
+# default = root
+[boot]
+# command = /usr/libexec/wsl-systemd
+# [automount]
+enable = true
+)"};
+            std::wstringstream fakeFile{buffer};
+            ASSERT_FALSE(ini_find_value(fakeFile, L"boot", L"command", L"/usr/libexec/wsl-systemd"));
+            ASSERT_FALSE(ini_find_value(fakeFile, L"user", L"default", L"root"));
+            ASSERT_FALSE(ini_find_value(fakeFile, L"automount", L"enabled", L"true"));
+        }
+        TEST(IniFindValueTests, WhiteSpacesAreIrrelevant)
+        {
+            std::wstring buffer{LR"(# This mimics the syntax of /etc/wsl.conf file with comments.
+[user]
+    default = root   
+[boot]
+    command = /usr/libexec/wsl-systemd
+)"};
+            std::wstringstream fakeFile{buffer};
+            ASSERT_TRUE(ini_find_value(fakeFile, L"boot", L"command", L"/usr/libexec/wsl-systemd"));
+            ASSERT_TRUE(ini_find_value(fakeFile, L"user", L"default", L"root"));
+        }
+        TEST(IniFindValueTests, IllFormedLinesStopParsing)
+        {
+            std::wstring buffer{LR"([automount]
+
+enabled = true
+mountFsTab = true
+
+[boot]
+command = /usr/libexec/wsl-systemd
+[user]
+name
+
+# The following configuration will never be applied because the syntax is broken in the line above. But the previous were.
+default = root
+)"};
+            std::wstringstream fakeFile{buffer};
+            ASSERT_TRUE(ini_find_value(fakeFile, L"boot", L"command", L"/usr/libexec/wsl-systemd"));
+            ASSERT_TRUE(ini_find_value(fakeFile, L"automount", L"enabled", L"true"));
+            ASSERT_TRUE(ini_find_value(fakeFile, L"automount", L"mountFsTab", L"true"));
+            ASSERT_FALSE(ini_find_value(fakeFile, L"user", L"default", L"root"));
+        }
+        TEST(IniFindValueTests, IllFormedSectionAlsoStopParsing)
+        {
+            std::wstring buffer{LR"([automount]
+enabled = true
+mountFsTab = true
+[network
+[boot]
+command = /usr/libexec/wsl-systemd
+)"};
+            std::wstringstream fakeFile{buffer};
+            ASSERT_TRUE(ini_find_value(fakeFile, L"automount", L"enabled", L"true"));
+            ASSERT_TRUE(ini_find_value(fakeFile, L"automount", L"mountFsTab", L"true"));
+            ASSERT_FALSE(ini_find_value(fakeFile, L"boot", L"command", L"/usr/libexec/wsl-systemd"));
+        }
+        TEST(IniFindValueTests, SectionsCanBeExtended)
+        {
+            std::wstring buffer{LR"([automount]
+enabled = true
+[boot]
+command = /usr/libexec/wsl-systemd
+[automount]
+mountFsTab = true
```

Comment:
Nice use cases :)

I was wondering if it worths adding some with empty section (no child value) and some with stripping spaces around =.

Reply:
Good that you asked that. It made me explore situations with spaces inside the section name. That seems to be interpreted as syntax error, since the configuration following such situations are not applied.

Now I'm pushing updates fixing this issue.